api/backend.go

```diff
@@ -166,6 +166,10 @@ func (b *StatusBackend) startNode(config *params.NodeConfig) (err error) {
 
 	signal.SendNodeReady()
 
+	if err := b.statusNode.StartDiscovery(); err != nil {
```

Comment:
If this can return an error, I would call it before `signal.SendNodeReady()`. It would be strange if this method fails but `node.ready` signal is sent.

Another option is starting the discovery mechanism in another thread and sending a signal if it fails (here is another question whether the UI should be bothered at all).

Reply:
Thanks for the explanation!